### PR TITLE
Update the logic for maintaining the telemetry preference

### PR DESCRIPTION
**Pull request by mybayern1974**
When there is version upgrade, if the existed version >= 3.x, then respect the recorded telemetry preference. Otherwise, overwrite it to "true";

### CODE REVIEW

**Comment by vscjavaci** [comment]
Can one of the admins verify this patch?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/AzurePlugin.java [hunk 1]
```diff
@@ -194,11 +196,16 @@ private void initializeAIRegistry() {
     }
 
     private void setValues(final String dataFile) throws Exception {
-        final Document doc = ParserXMLUtility.parseXMLFile(dataFile);
-        DataOperations.updatePropertyValue(doc, message("prefVal"), String.valueOf("true"));
-        DataOperations.updatePropertyValue(doc, message("pluginVersion"), PLUGIN_VERSION);
-        DataOperations.updatePropertyValue(doc, message("instID"), _hashmac);
         try {
+            final Document doc = ParserXMLUtility.parseXMLFile(dataFile);
+            String recordedVersion = DataOperations.getProperty(dataFile, message("pluginVersion"));
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/AzurePlugin.java]: get Property on dataFile or doc?

File: Utils/azuretools-core/src/com/microsoft/azuretools/azurecommons/util/Utils.java [hunk 2]
```diff
@@ -154,4 +154,54 @@ public static boolean isPortAvailable(int port) throws Exception {
...
+        String[] version1Seg = version1.split("\\.");
+        String[] version2Seg = version2.split("\\.");
+
+        int i = 0;
+        while (i < version1Seg.length && i < version2Seg.length && version1Seg[i].equalsIgnoreCase(version2Seg[i])) {
+            i++;
+        }
+
+        if (i < version1Seg.length && i < version2Seg.length) {
+            return Integer.valueOf(version1Seg[i]).compareTo(Integer.valueOf(version2Seg[i]));
```
Comment (adashen) [inline Utils/azuretools-core/src/com/microsoft/azuretools/azurecommons/util/Utils.java]: Integer.valueOf will throw exception, need handle the exception

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/AzurePlugin.java (hunk 1 above)
Comment (mybayern1974) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/AzurePlugin.java]: on dataFile. The logic has been tested.

File: Utils/azuretools-core/src/com/microsoft/azuretools/azurecommons/util/Utils.java (hunk 2 above)
Comment (mybayern1974) [inline Utils/azuretools-core/src/com/microsoft/azuretools/azurecommons/util/Utils.java]: The exception is handled at the callers that call this function. The exceptional test cases have been verified.

**Comment by Eskibear** [comment]
retest this please

**Comment by testforstephen** [comment]
add to whitelist

**Pull request merged**